modules/bgsegm/src/bgfg_subcnt.cpp

```diff
@@ -0,0 +1,429 @@
+//
+//  IMPORTANT: READ BEFORE DOWNLOADING, COPYING, INSTALLING OR USING.
+//
+//  By downloading, copying, installing or using the software you agree to this license.
+//  If you do not agree to this license, do not download, install,
+//  copy or use the software.
+//
+//
+//                          License Agreement
+//                       (3-clause BSD License)
+//                     For BackgroundSubtractorCNT
+//               (Background Subtraction based on Counting)
+//
+// Copyright (C) 2016, Sagi Zeevi (www.theimpossiblecode.com), all rights reserved.
+// Third party copyrights are property of their respective owners.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+//   * Redistribution's of source code must retain the above copyright notice,
+//     this list of conditions and the following disclaimer.
+//
+//   * Redistribution's in binary form must reproduce the above copyright notice,
+//     this list of conditions and the following disclaimer in the documentation
+//     and/or other materials provided with the distribution.
+//
+//   * The name of the copyright holders may not be used to endorse or promote products
+//     derived from this software without specific prior written permission.
+//
+// This software is provided by the copyright holders and contributors "as is" and
+// any express or implied warranties, including, but not limited to, the implied
+// warranties of merchantability and fitness for a particular purpose are disclaimed.
+// In no event shall the Intel Corporation or contributors be liable for any direct,
+// indirect, incidental, special, exemplary, or consequential damages
+// (including, but not limited to, procurement of substitute goods or services;
+// loss of use, data, or profits; or business interruption) however caused
+// and on any theory of liability, whether in contract, strict liability,
+// or tort (including negligence or otherwise) arising in any way out of
+// the use of this software, even if advised of the possibility of such damage.
+//
+//M*/
+
+
+#include "precomp.hpp"
+#include <functional>
+
+namespace cv
+{
+namespace bgsegm
+{
+
+/** @brief Implementation of background subtraction based on counting.
+ *  About as fast as MOG2 on a high end system (benchmarked on )
+ *  More than twice faster than MOG2 on cheap hardware (benchmarked on Raspberry Pi3).
+ *  Algorithm by Sagi Zeevi
+ */
+class BackgroundSubtractorCNTImpl: public BackgroundSubtractorCNT
+{
+public:
+    /**
+     * @brief BackgroundSubtractorCNTImpl
+     * @param stability number of frames with same pixel color to consider stable
+     * @param useHistory determines if we're giving a pixel credit for being stable for a long time
+     * @param maxStability maximum allowed credit for a pixel in history
+     * @param isParallel determines if we're parallelizing the algorithm
+     */
```

Comment:
This documentation comment should be moved into .hpp file for `createBackgroundSubtractorCNT` function.